From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 23 Jan 2016 23:24:59 +0000
Subject: af_unix: Avoid ABI changes
Forwarded: not-needed

struct unix_sock is only allocated in af_unix so it's safe to add new
members at the end.  Hide gc_flags and peer_wake from genksyms.

Include padding in place of the bitfields that were removed.  These are
not accessed outside of the af_unix implementation.
---
--- a/include/net/af_unix.h
+++ b/include/net/af_unix.h
@@ -54,12 +54,20 @@ struct unix_sock {
 	struct list_head	link;
 	atomic_long_t		inflight;
 	spinlock_t		lock;
+#ifdef __GENKSYMS__
+	unsigned int		gc_candidate : 1;
+	unsigned int		gc_maybe_cycle : 1;
+#else
+	unsigned int		pad : 2;
+#endif
 	unsigned char		recursion_level;
+	struct socket_wq	peer_wq;
+#ifndef __GENKSYSMS__
+	wait_queue_t		peer_wake;
 	unsigned long		gc_flags;
 #define UNIX_GC_CANDIDATE	0
 #define UNIX_GC_MAYBE_CYCLE	1
-	struct socket_wq	peer_wq;
-	wait_queue_t		peer_wake;
+#endif
 };
 #define unix_sk(__sk) ((struct unix_sock *)__sk)
 
